src/nnet3/nnet-test-utils.cc

```diff
@@ -1026,12 +1026,21 @@ static void GenerateRandomComponentConfig(std::string *component_type,
     case 24: {
       *component_type = "RepeatedAffineComponent";
       int32 num_repeats = RandInt(1, 50),
-          input_dim = num_repeats * RandInt(1, 15),
-          output_dim = num_repeats * RandInt(1, 15);
```

Comment:
The numerical routines (i.e., Propagate and Backprop) did not crash when input_dim=1 and output_dim=1. Rather, the crash in ./nnet-component-test was because predicted_obj_change was not approximately equal to measured_objf_change in TestSimpleComponentDataDerivative().

Obviously, the test works for other (larger) input parameters. Simply removing a test case that doesn't work seems a little bad, but I think the test is likely to be a false positive since the tests work for other tested parameters.

Of course I also don't really understand the tests as well as I should. (I don't understand why the output_deriv-output dot product is a valid objective function.)